pyopia/pipeline.py

```diff
@@ -0,0 +1,56 @@
+class Pipeline():
+    '''Processing pipeline class
+
+    further explanation @todo
+    '''
+
+    def __init__(self, steps):
+        self.steps = steps
+        self.cl = self.steps['classifier']()
+        self.cl.load_model()
+
+    def run(self):
+        '''Method for executing the processing pipeline
+
+        Returns:
+            stats (DataFrame): stats DataFrame of particle statistics
+        '''
+
+        timestamp, imc = self.steps['load']()
+
+        stats = self.steps['statextract'](timestamp, imc, self.cl)
```

Comment:
how do you think we can solve this? I guess it is an inbuilt disadvantage of the design?

Reply:
Yeah, might just be an inbuilt disadvantage. Maybe having a python script with an example helps a bit (instead of a jupyter notebook), but that creates duplicate code. I guess for testing there will have to be a python script that runs the silcam pipeline/workflow, or some particular pipeline/workflow you care about.